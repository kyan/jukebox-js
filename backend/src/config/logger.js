import { Logger } from 'tslog'

const logStuff = new Logger()

export default logStuff
