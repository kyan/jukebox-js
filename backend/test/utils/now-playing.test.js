import Setting from '../../src/models/setting'
import logger from '../../src/config/logger'
import NowPlaying from '../../src/utils/now-playing'
jest.mock('../../src/config/logger')

describe('NowPlaying', () => {
  describe('addTrack', () => {
    const trackObject = {
      name: 'Seasons (Waiting On You)',
      year: '1983',
      image: 'the-album-art.jpg',
      artist: {
        name: 'Future Islands'
      },
      album: {
        name: 'Singles'
      },
      metrics: {
        votesAverage: 80,
        votes: 2,
        plays: 2
      },
      addedBy: [
        {
          user: {
            fullname: 'Duncan'
          },
          addedAt: new Date(1582010703141)
        },
        {
          user: {
            fullname: 'BRH'
          },
          addedAt: new Date(1582000703141)
        }
      ]
    }

    it('returns the correct payload when full data', () => {
      jest.spyOn(Setting, 'findOneAndUpdate').mockResolvedValue()
      jest.spyOn(global.Date, 'now').mockImplementation(() => 1582020703141)

      return NowPlaying.addTrack(trackObject)
        .then(payload => {
          expect(payload).toMatchSnapshot()
        })
    })

    it('returns the correct payload when full data 1', () => {
      jest.spyOn(Setting, 'findOneAndUpdate').mockResolvedValue()
      jest.spyOn(global.Date, 'now').mockImplementation(() => 1582020703141)
      trackObject.metrics.plays = 1
      delete trackObject.addedBy[1]

      return NowPlaying.addTrack(trackObject)
        .then(payload => {
          expect(payload).toMatchSnapshot()
        })
    })

    it('handles errors', () => {
      jest.spyOn(Setting, 'findOneAndUpdate').mockRejectedValue(new Error('oooops'))
      NowPlaying.addTrack(trackObject)

      return new Promise(resolve => {
        setTimeout(() => {
          expect(logger.error).toHaveBeenCalledWith('NowPlaying.addTrack: oooops')
          resolve()
        }, 0)
      })
    })
  })
})
