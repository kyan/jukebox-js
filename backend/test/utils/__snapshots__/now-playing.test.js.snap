// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NowPlaying addTrack returns the correct payload when full data 1`] = `
Object {
  "blocks": Array [
    Object {
      "text": Object {
        "text": "Now Playing:",
        "type": "mrkdwn",
      },
      "type": "section",
    },
    Object {
      "type": "divider",
    },
    Object {
      "accessory": Object {
        "alt_text": "Future Islands",
        "image_url": "the-album-art.jpg",
        "type": "image",
      },
      "block_id": "section1",
      "text": Object {
        "text": "*Seasons (Waiting On You)*
 Future Islands
 Singles (1983)",
        "type": "mrkdwn",
      },
      "type": "section",
    },
    Object {
      "type": "divider",
    },
    Object {
      "block_id": "section2",
      "fields": Array [
        Object {
          "text": "*Rating:* 3",
          "type": "mrkdwn",
        },
        Object {
          "text": "*Voted by:* 2 users",
          "type": "mrkdwn",
        },
        Object {
          "text": "*Played:* 2 times",
          "type": "mrkdwn",
        },
        Object {
          "text": "*Last Played:* 6 hours ago",
          "type": "mrkdwn",
        },
      ],
      "type": "section",
    },
    Object {
      "elements": Array [
        Object {
          "text": "Added by Duncan 3 hours ago",
          "type": "mrkdwn",
        },
      ],
      "type": "context",
    },
  ],
}
`;

exports[`NowPlaying addTrack returns the correct payload when full data 1 1`] = `
Object {
  "blocks": Array [
    Object {
      "text": Object {
        "text": "Now Playing:",
        "type": "mrkdwn",
      },
      "type": "section",
    },
    Object {
      "type": "divider",
    },
    Object {
      "accessory": Object {
        "alt_text": "Future Islands",
        "image_url": "the-album-art.jpg",
        "type": "image",
      },
      "block_id": "section1",
      "text": Object {
        "text": "*Seasons (Waiting On You)*
 Future Islands
 Singles (1983)",
        "type": "mrkdwn",
      },
      "type": "section",
    },
    Object {
      "type": "divider",
    },
    Object {
      "block_id": "section2",
      "fields": Array [
        Object {
          "text": "*Rating:* 3",
          "type": "mrkdwn",
        },
        Object {
          "text": "*Voted by:* 2 users",
          "type": "mrkdwn",
        },
        Object {
          "text": "*Played:* 1 time",
          "type": "mrkdwn",
        },
      ],
      "type": "section",
    },
    Object {
      "elements": Array [
        Object {
          "text": "Added by Duncan 3 hours ago",
          "type": "mrkdwn",
        },
      ],
      "type": "context",
    },
  ],
}
`;
